src/commands.cpp

```diff
@@ -5715,15 +5737,28 @@ void CommandQueryRecoveryLink::procresult()
 
     client->json.enterarray();
 
-    int type = API_EINTERNAL;
     string email;
     string ip;
     m_time_t ts;
     handle uh;
 
-    if (!client->json.isnumeric() || ((type = int(client->json.getint())) < 0))   // error
+    int64_t type;
+    Error errorDetails;
+
+    if (checkError(errorDetails, client->json) && errorDetails < 0)
     {
-        return client->app->queryrecoverylink_result((error)type);
+        return client->app->queryrecoverylink_result(errorDetails);
+    }
+
+    type = static_cast<uint64_t>(errorDetails);
```

Comment:
Why does the data type of `type` changes from `int` to `int64_t`??
And why the casting to `uint64_t`??
I think we can simply keep the `int`

Reply:
Because it can be negative, if it will be negative, it would be considerate as an error. But it's the same